python/cudf/cudf/indexing.py

```diff
@@ -68,20 +69,31 @@ def __getitem__(self, arg):
 
 class _DataFrameIndexer(object):
     def __getitem__(self, arg):
-        if type(arg) is not tuple:
-            arg = (arg, slice(None, None))
-
-        if self._is_multiindex_arg(arg):
-            return self._getitem_multiindex_arg(arg)
-        elif self._is_scalar_access(arg):
-            return self._getitem_scalar(arg)
-
+        arg = self._create_valid_tuple(arg)
         df = self._getitem_tuple_arg(arg)
 
-        if self._can_downcast_to_series(df, arg):
```

Comment:
Any reason for pushing this down into `_getitem_tuple_arg`? It gives the reader the impression that we always return a `DataFrame`.

Reply:
I might be able to pull it back up. The reason is because there are many "downcast" type operations that can occur during getitem_tuple_arg, and shuffling them all to occur after `get` will be difficult.

Reply:
The issue has to do with different rules for downcasting to series depending on the shape of the multiindex, or if .iloc or .loc are used to access. I doubt I can pull it up. :D

Reply:
OK. Then I'd suggest:

```
series_or_df = self._getitem_tuple_arg(arg)
```

simply because `df` is misleading

Reply:
👍 I was leaning toward just doing something like "result", but I'll be more specific if you like. :)